mlir/lib/Dialect/Rock/Transforms/FoldTranspose.cpp

```diff
@@ -39,6 +41,207 @@ using namespace mlir;
 using namespace mlir::rock;
 
 namespace {
+
+// This rewrite will rewrite the linalg IO that has view like-ops surrounding
+// them to be consumed by the linalg operation itself adjusting the indexing
+// maps to faithfully represent them.
+struct InlineViewLikeOperandsLinalgRewritePattern
+    : public OpRewritePattern<linalg::GenericOp> {
+  using OpRewritePattern<linalg::GenericOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(linalg::GenericOp laGeneric,
+                                PatternRewriter &rewriter) const override;
+};
+
+// This function will take strides of the original layout of corresponding
+// dimensions being reassociated and the expected resultant dim sizes of
+// post-reassociation --> then, produce the set of resultant AffineExprs after
+// the reassociation.
+static void getDelinearizedAffineExpr(
+    ArrayRef<int64_t> originalStridesDimsBeingReassociated,
+    ArrayRef<int64_t> postReassociatedDimSizes, Builder &b,
+    unsigned int position, SmallVectorImpl<mlir::AffineExpr> &res) {
```

Comment:
The `mlir::` nit

Reply:
Ah missed that.. I ll do a follow up on Monday... Taking this in for now coz of potential conflicts as Im out today.